PHP extension integration

asgrim opened the pull request:
Fixes #41

cschneid commented on composer.json:
```diff
@@ -11,7 +11,8 @@
         "ralouphie/getallheaders": "2.0.5",
-        "ramsey/uuid": "^3.7"
+        "ramsey/uuid": "^3.7",
+        "webmozart/assert": "^1.4"
```
Can this be in require-dev instead so it doesn't conflict with anything another version a user may have required too? Seems to be only used in tests.

asgrim replied:
It can't since it's used in `\Scoutapm\Extension\RecordedCall` - however, we can be compatible with `^1.0` so I'll loosen the requirement.

asgrim commented:
@cschneid this is ready for review now please (note: please don't squash when merging, there are some potentially useful scripts that I added and removed within this PR which may be useful to refer back to later)

cschneid commented on src/Extension/RecordedCall.php:
```diff
@@ -0,0 +1,70 @@
+        Assert::keyExists($extensionCall, 'function');
+        Assert::keyExists($extensionCall, 'entered');
+        Assert::keyExists($extensionCall, 'exited');
+        Assert::keyExists($extensionCall, 'time_taken');
```
Does php not have a builtin for has_key?

asgrim replied:
Yes, [`array_key_exists`](https://www.php.net/manual/en/function.array-key-exists.php) serves this purpose (and is actually used by the assertion library). The assertion library just abstracts away the logical assertion into a single semantic statement, the following is more or less equivalent:

```php
if (!isset($extensionCall['time_taken']) || !array_key_exists('time_taken', $extensionCall)) {
  throw new \InvalidArgumentException('The key time_taken did not exist');
}
```
